cblite/llm/Gemini.cc

```diff
@@ -22,25 +22,69 @@ using namespace std;
 using namespace fleece;
 using namespace litecore;
 
-alloc_slice Gemini::run(Value rawSrcPropValue, const string& modelName) {
-    string restBody = format("{\"model\": \"models/%s\", \"content\": {\"parts\": [{\"text\": \"%.*s\"}]}}", modelName.c_str(), SPLAT(rawSrcPropValue.asString()));
+vector<alloc_slice> Gemini::run(const string& modelName, vector<Value> propVec) {
+    int batchSize = 50, docRemainder = propVec.size() % batchSize;
+    long batches = propVec.size() / batchSize;
+    if (propVec.size() % batchSize != 0) {
+        batches++;
+    }
+    vector<alloc_slice> responses;
+    alloc_slice response, completeResponse;
+    Value rawSrcPropValue;
     
-    // Get headers
-    Encoder enc;
-    enc.beginDict();
-    enc["Content-Type"_sl] = "application/json";
-    enc["x-goog-api-key"_sl] = getenv("LLM_API_KEY");
-    enc.endDict();
-    auto headers = enc.finishDoc();
-    
-    // Run request
-    auto r = std::make_unique<REST::Response>("https", "POST", "generativelanguage.googleapis.com", 443, format("v1beta/models/%s:embedContent", modelName.c_str()));
-    r->setHeaders(headers).setBody(restBody);
-    return LLMProvider::run(r);
+    for (int i = 0; i < batches; i++) {
+        // Create rest body
+        string restBody = format("{\"requests\": [");
+        if ((i == batches - 1) && (docRemainder != 0)) {
+            int j = 0;
+            while (j < docRemainder) {
+                rawSrcPropValue = propVec.at(j);
+                restBody += format(" {\"model\": \"models/%s\", \"content\": {\"parts\": [{\"text\": \"%.*s\"}]}, },", modelName.c_str(), SPLAT(rawSrcPropValue.asString()));
+                j++;
+            }
+            propVec.erase(propVec.begin(), propVec.begin() + (docRemainder - 1));
+        } else {
+            for (int k = 0; k < batchSize; k++) {
+                rawSrcPropValue = propVec.at(k);
+                restBody += format(" {\"model\": \"models/%s\", \"content\": {\"parts\": [{\"text\": \"%.*s\"}]}, },", modelName.c_str(), SPLAT(rawSrcPropValue.asString()));
+            }
+            propVec.erase(propVec.begin(), propVec.begin() + (batchSize - 1));
+        }
+        restBody += "]}";
+        
+        // Get headers
+        Encoder enc;
+        enc.beginDict();
+        enc["Content-Type"_sl] = "application/json";
+        enc.endDict();
+        auto headers = enc.finishDoc();
+        
+        // Run request
+        cout << "Running batch " << i + 1;
```

Comment:
This output might be kind of confusing to look at.  Perhaps batch x of request y?